core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -87,20 +88,65 @@
    private StateTransferLock stateTransferLock;
    private XSiteStateConsumer xSiteStateConsumer;
    private GroupManager groupManager;
+   private CacheNotifier notifier;
 
    private static final Log log = LogFactory.getLog(EntryWrappingInterceptor.class);
    private static final boolean trace = log.isTraceEnabled();
    private static final EnumSet<Flag> EVICT_FLAGS =
          EnumSet.of(Flag.SKIP_OWNERSHIP_CHECK, Flag.CACHE_MODE_LOCAL);
 
+   private final ReturnHandler dataReadReturnHandler = new ReturnHandler() {
```

Comment:
I actually prefer the inner classes because I'm logging the return handler class names in `BaseAsyncInvocationContext` and it's a bit easier to find the inner class definition compared to the lambda. Of course, tooling is bound to improve this, so I'll give it a try after I'm done with the other comments (and check the performance with `::` method references, as well).